src/bedrock/kvs/utils.cpp

```diff
@@ -46,3 +46,25 @@ void process_put(const Key& key, const unsigned long long& timestamp,
     key_size_map[key] = value.size();
   }
 }
+
+bool is_primary_replica(const Key& key, std::unordered_map<Key, KeyInfo>& placement,
+                        std::unordered_map<unsigned, GlobalHashRing>& global_hash_ring_map,
+                        std::unordered_map<unsigned, LocalHashRing>& local_hash_ring_map,
+                        ServerThread& st) {
+  if (placement[key].global_replication_map_[kSelfTierId] == 0) {
+    return false;
+  } else if (kSelfTierId == 2 && placement[key].global_replication_map_[1] > 0) {
```

Comment:
We shouldn't hard code 2-tier specifics into this, especially if we want to add a shared memory tier soon. A better check is probably:
```
else if (kSelfTierId > 1 
            && placement[key].global_replication_map_[any_tier_id < kSelfTierId] > 0) 
{ ... }
```

Reply:
made the change